client/purplecaffeine/core.py

```diff
@@ -465,12 +465,26 @@ def list(
         offset = offset or 0
         limit = limit or 10
 
-        trails_path = glob.glob(f"{self.path}/**.json")[offset:limit]
+        trials_path = glob.glob(f"{self.path}/**.json")
         trials = []
-        for path in trails_path:
+        for path in trials_path:
             with open(path, "r", encoding="utf-8") as trial_file:
                 trial_dict = json.load(trial_file, cls=TrialDecoder)
                 trials.append(Trial(**trial_dict))
+
+        if query:
+            trials = [
+                trial
+                for trial in trials
+                if (query in trial.tags)
+                or (trial.name.find(query) != -1)
+                or (trial.description.find(query) != -1)
+            ]
+            if len(trials) == 0:
+                logging.warning("No Trials returned for the query '%s'.", query)
```

Comment:
I think we should return empty results if nothing has been found :) 

Error will scare people :)